### PR TITLE
6452 local dev instructions tweaks

### PR DESCRIPTION
**Pull request by cnburnett**
Please review changes suggested for two README.md files.

See the PR testing procedure in https://apps.itsm.gov.bc.ca/jira/browse/GDXDSD-6452

In short, best testing would be by a dev without the apps and dependencies loaded.
1. (As needed) Review READMEs and remove apps and dependencies
2.  Follow instructions in backend and frontend READMEs

### CODE REVIEW

File: backend/README.md [hunk 1]
```diff
@@ -1,38 +1,29 @@
...
+# URL-S Backend
 
-## Description
+Instructions for configuring and starting the URL-S backend apps: Node.js Express and MongoDB.
 
-This is an example backend app that authenticates users using OAuth 2.0 standard flow
+## Local Setup
+ **`Step 1`**. Make sure you have Node.js installed in your local machine. Type node -v in Terminal. This repo has been tested with Node version 20.10.0 LTS. If you do not have Node, download from here - https://nodejs.org/en/. npm will be installed alongside node.js.
+ 
+ **`Step 2`**. Make sure you have MongoDB installed and running. Install Homebrew and then `brew install mongodb-community@7.0`. Start the database service with `brew services start mongodb-community`
```
Comment (veenupunyani) [inline backend/README.md:8]: We need this command before we can install the community edition
`brew tap mongodb/brew`

Also Compass can be installed visa brew 
`brew install --cask mongodb-compass`

File: backend/README.md [hunk 2]
```diff
@@ -1,38 +1,29 @@
...
+ 
+ **`Step 3`**. Make sure you have the MongoDB Compass (GUI) app available. This desktop app is used to manage MongoDB databases. Find it at https://www.mongodb.com/try/download/compass. Check that you can connect to the database at localhost:27017 (mongodb://localhost:27017).
```
Comment (veenupunyani) [inline backend/README.md:10]: No need for this step if compass has already been installed via brew in step-2

File: backend/README.md [hunk 3]
```diff
@@ -1,38 +1,29 @@
...
+ 
+ **`Step 4`**. Create an .env file to store the BC Government Single Sign-on SSO secret. Copy .env.template to .env. Download the secret from https://bcgov.github.io/sso-requests; click Login; select the URL Shortener project; click Download in INTEGRATION DETAILS; Open the downloaded .json file in a text editor. Copy and paste the secret value into the new .env file. Note that .gitignore lists .env - it will not be added to the repository.
```
Comment (veenupunyani) [inline backend/README.md:12]: It wasn't really clear where to put the .env file. It should be mentioned that it needs to be put under backend folder.

File: backend/README.md [hunk 4]
```diff
@@ -1,38 +1,29 @@
...
+ 
+ **`Step 5`**. From the /backend folder, run `npm install` in the terminal to install required dependencies.
+ 
+ **`Step 6`**. From the /backend folder, run `npm run start` in the terminal to start the backend app.
 
-## Getting Started
+**`Step 7`**. Open a browser and visit http://localhost:3000/. Click 'Proceed to login...' to enter your IDIR credentials
 
-## Pre-requisites
+**`Step 8`**. Complete the the Frontend installation steps in the /frontend folder README.
```
Comment (veenupunyani) [inline backend/README.md:20]: We should add here to keep this terminal running and open a new terminal for frontend

File: frontend/README.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# URL-S Frontend
+
+Instructions for configuring and starting the URL-S frontend app, Vue.js, and its dependencies.
+
+It is recommended that you install and start the Backend applications before using the Frontend. See the README in the Backend folder.
```
Comment (veenupunyani) [inline frontend/README.md:5]: recommended should be changed to necessary as it is a dependent step.

File: frontend/README.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+## Local Setup
+**`Step 1`**. Open a terminal (option to do this in MS Code). Change directory to the frontend/url-shortener folder
```
Comment (veenupunyani) [inline frontend/README.md:8]: lets be clear on the folder here as at a first glance i tried to do the npm install in front-end folder, but it needs to be done one folder inside it.

Comment (veenupunyani) [inline frontend/README.md:8]: Ask to open a new terminal for frontend opreations

File: frontend/README.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  
+**`Step 2`**. From the /url-shortener folder, run `npm install` in the terminal to install required dependencies.
+ 
+**`Step 3`**. From the /url-shortener folder, run `npm run dev` in the terminal to start the app.
+
+**`Step 4`**. Open a browser and visit the app dev site at http://localhost:5173
```
Comment (veenupunyani) [inline frontend/README.md:14]: We should add here on testing the app once it is installed. Like how to check the values in the database to confirm the flow

**Review by veenupunyani — CHANGES_REQUESTED** [review]
hi @cnburnett i have finished my review, these are my recommended changes. let me know if you want to add the changes or want me to do it as it is your branch so i don't want to touch it if you don't like

File: frontend/README.md (hunk 5 above)
Comment (cnburnett) [reply frontend/README.md]: I added text in Commit d16d4d9aecf196fa52aaebfe84b56646563df241

"Before doing the the frontend install and start, the Backend application must be installed and started. Follow instructions in..."

File: backend/README.md (hunk 1 above)
Comment (cnburnett) [reply backend/README.md:8]: Fixed with commit = "Update README for mongodb/brew and grew Compass changes"

File: backend/README.md (hunk 2 above)
Comment (cnburnett) [reply backend/README.md:10]: This is covered in commit = Update README for mongodb/brew and grew Compass changes

File: backend/README.md (hunk 3 above)
Comment (cnburnett) [reply backend/README.md:12]: Fixed in commit = Update README.md to clarify env file must be created in the backend fold

File: frontend/README.md (hunk 6 above)
Comment (cnburnett) [reply frontend/README.md]: Fixed in commit = Update README.md to clarify env file must be created in the backend folder

File: backend/README.md (hunk 4 above)
Comment (cnburnett) [reply backend/README.md:20]: Fixed in commit = Update README.md to clarify env file must be created in the backend folder

File: frontend/README.md (hunk 7 above)
Comment (cnburnett) [reply frontend/README.md:14]: @veenupunyani I cannot do this quickly. I will make a new ticket to add testing steps to the READMEs

**Comment by cnburnett** [comment]
Hi @veenupunyani back to you for a follow-up review and approval.

**Review by veenupunyani — APPROVED** [review]
Thanks for making those changes

**Pull request merged**